vertx-consul-client/src/main/java/io/vertx/ext/consul/ConsulClientOptions.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * Copyright (c) 2016 The original author or authors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Apache License v2.0 which accompanies this distribution.
+ *
+ *      The Eclipse Public License is available at
+ *      http://www.eclipse.org/legal/epl-v10.html
+ *
+ *      The Apache License v2.0 is available at
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * You may elect to redistribute this code under either of these licenses.
+ */
+package io.vertx.ext.consul;
+
+import io.vertx.codegen.annotations.DataObject;
+import io.vertx.core.json.JsonObject;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PemTrustOptionsConverter;
+
+/**
+ * Options used to create Consul client.
+ *
+ * @author <a href="mailto:ruslan.sennov@gmail.com">Ruslan Sennov</a>
+ */
+@DataObject(generateConverter = true)
+public class ConsulClientOptions {
+
+  private String host;
+  private int port;
+  private String aclToken;
+  private String dc;
+  private long timeoutMs;
+  private boolean ssl;
+  private boolean trustAll;
+  private PemTrustOptions pemTrustOptions;
+
+  /**
+   * Default constructor
+   */
+  public ConsulClientOptions() {
+  }
+
+  /**
+   * Copy constructor
+   *
+   * @param options the one to copy
+   */
+  public ConsulClientOptions(ConsulClientOptions options) {
+    this.host = options.host;
+    this.port = options.port;
+    this.aclToken = options.aclToken;
+    this.dc = options.dc;
+    this.timeoutMs = options.timeoutMs;
+    this.ssl = options.ssl;
+    this.pemTrustOptions = options.pemTrustOptions;
+  }
+
+  /**
+   * Constructor from JSON
+   *
+   * @param options the JSON
+   */
+  public ConsulClientOptions(JsonObject options) {
+    ConsulClientOptionsConverter.fromJson(options, this);
+  }
+
+  /**
+   * Convert to JSON
+   *
+   * @return the JSON
+   */
+  public JsonObject toJson() {
+    JsonObject jsonObject = new JsonObject();
+    ConsulClientOptionsConverter.toJson(this, jsonObject);
+    if (pemTrustOptions != null) {
+      JsonObject pem = new JsonObject();
+      PemTrustOptionsConverter.toJson(pemTrustOptions, pem);
+      jsonObject.put("pemTrustOptions", pem);
```

Comment:
I think it would be better and simpler if your options inherit or embed HttpClientOptions, there are other bunch of things we will get like client proxy support, and TCP options